cpp/src/arrow/json/parser.h

```diff
@@ -56,7 +56,8 @@ struct Kind {
   static Status ForType(const DataType& type, Kind::type* kind);
 };
 
-constexpr int32_t kMaxParserNumRows = 100000;
+/// Set the max parse row is 2147483647.
+constexpr int32_t kMaxParserNumRows = std::numeric_limits<int32_t>::max();
 
```

Comment:
IMO, it would be slightly more readable to use the limit directly below and modify the error return from "Exceeded maximum rows" to "Row count overflowed int32_t"
```suggestion
```

Reply:
Hi @bkietz, thanks for your review. The error msg has update. 
I personally prefer to define the result of `int32_t::max()` as a constant in `parse.h` to be consistent with other constants. 
Do you suggest moving it to `parse.cc` or direct using it like `for (; num_rows_ < [kMaxParserNumRows](std::numeric_limits<int32_t>::max()); ++num_rows_) {}` on `parse.cc` ?

Reply:
I think it's more readable with direct usage in this case; since the error message now explains that we are guarding against int32 overflow, we should make it clear inline that we are  checking against int32's max. A named constant can be introduced when we're limiting against something configurable instead of the numeric limits of int32.

Reply:
> I think it's more readable with direct usage in this case; since the error message now explains that we are guarding against int32 overflow, we should make it clear inline that we are checking against int32's max. A named constant can be introduced when we're limiting against something configurable instead of the numeric limits of int32.

Thank you for your explanation, I have removed the constant and now using it directly in the loop.